src/install/lockfile.zig

```diff
@@ -3853,3 +3658,681 @@ pub fn resolve(this: *Lockfile, package_name: []const u8, version: Dependency.Ve
+        return advanceUntilAnyInternal(cur, chars_, advance_over_current, .naive);
+    }
+
+    pub fn advanceUntilAny(cur: *[]const u8, comptime chars_: []const u8, comptime advance_over_current: bool) !void {
+        return advanceUntilAnyInternal(cur, chars_, advance_over_current, ADVANCE_IMPL);
+    }
+
+    inline fn advanceUntilAnyInternal(cur: *[]const u8, comptime chars_: []const u8, comptime advance_over_current: bool, impl: @TypeOf(ADVANCE_IMPL)) !void {
+        comptime std.debug.assert(chars_.len > 0);
+        comptime var chars = chars_;
+        const SHOULD_ERROR_ON_BACKSLASH = comptime std.mem.indexOfAny(u8, chars, "\"") != null;
+
+        // check at comptime to make sure this is safe
+        comptime if (std.mem.indexOfAny(u8, &SENTINELS, chars) == null)
+            @compileError(std.fmt.comptimePrint(
+                "Not guaranteed to terminate! Please place {s} at the end of `SENTINELS`.",
+                .{if (chars.len > 1) std.fmt.comptimePrint("one of [{s}]", .{chars}) else chars},
+            ));
+
+        if (SHOULD_ERROR_ON_BACKSLASH) {
+            chars = chars ++ "\\";
+        }
+
+        if (impl != .naive and advance_over_current) cur.* = cur.*[1..];
+        const width = ADVANCE_WIDTH;
+        comptime std.debug.assert(over_allocated_space >= width);
+
+        switch (impl) {
+            .naive => OUTER: while (true) {
+                if (advance_over_current) cur.* = cur.*[1..];
+                const c = cur.*[0];
+
+                inline for (chars) |chr| {
+                    if (c == chr) break :OUTER;
+                }
+
+                if (!advance_over_current) cur.* = cur.*[1..];
+            },
+
+            .SWAR => while (true) {
+                const word = @bitCast(u64, cur.*[0..width].*);
+
+                // Algorithm from here: http://0x80.pl/notesen/2023-03-06-swar-find-any.html
+                comptime var one_in_each_byte: std.meta.Int(.unsigned, width * 8) = 0;
+                comptime for (0..width) |i| {
+                    one_in_each_byte |= 1 << (i * 8);
+                };
+
+                const mask: usize = one_in_each_byte * 0x7F;
+                const mask2: usize = one_in_each_byte * 0x80;
+                const lo7bits: usize = word & mask;
+                var t0: usize = std.math.maxInt(usize);
+
+                inline for (chars) |chr| {
+                    comptime if (chr >= 0x80)
+                        @compileError("advanceUntilAny can only search for ascii characters in its SWAR implementation.");
+
+                    t0 &= (lo7bits ^ (comptime chr * one_in_each_byte)) + mask;
+                }
+
+                const match_mask = ((t0 | word) & mask2) ^ mask2;
+                cur.* = cur.*[@ctz(match_mask) / 8 ..]; // do a ctz no matter what, avoid the branch
+                if (match_mask != 0) break;
+            },
```

Comment:
I am not sure if this code will break on Big Endian machines.

Reply:
It 100% doesn’t matter 

We won’t support big endian for awhile